packages/interactivity/src/directives.tsx

```diff
@@ -354,19 +354,24 @@ export default () => {
 				.forEach( ( entry ) => {
 					const className = entry.suffix;
 					const result = evaluate( entry );
-					const currentClass = element.props.class || '';
-					const classFinder = new RegExp(
-						`(^|\\s)${ className }(\\s|$)`,
-						'g'
-					);
+
+					const classProxyElement = document.createElement( 'div' );
+					if ( 'string' === typeof element.props.class ) {
+						classProxyElement.className = element.props.class;
+					}
+
 					if ( ! result ) {
-						element.props.class = currentClass
-							.replace( classFinder, ' ' )
-							.trim();
-					} else if ( ! classFinder.test( currentClass ) ) {
-						element.props.class = currentClass
-							? `${ currentClass } ${ className }`
-							: className;
+						if (
+							classProxyElement.classList.contains( className )
+						) {
+							classProxyElement.classList.remove( className );
+							element.props.class = classProxyElement.className;
+						}
+					} else if (
+						! classProxyElement.classList.contains( className )
+					) {
+						classProxyElement.classList.add( className );
+						element.props.class = classProxyElement.className;
 					}
```

Comment:
Should we destroy the `classProxyElement` here to prevent leaking memory?

Reply:
Or perhaps we could create the `classProxyElement` in the upper scope and reuse
the same element every time the `class` directive processor runs.

Reply:
I don't think there's a leak here, this variable is created, moves out of scope, and should be eligible for garbage collection at that point.

Re-using the element may be a good idea regardless, I'll explore that.